core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -537,12 +537,16 @@ class UnifiedLog(@volatile var logStartOffset: Long,
   newGauge(LogMetricNames.LogEndOffset, () => logEndOffset, tags)
   newGauge(LogMetricNames.Size, () => size, tags)
 
-  val producerExpireCheck = scheduler.schedule(name = "PeriodicProducerExpirationCheck", fun = () => {
+  val producerExpireCheckOption = scheduler.schedule(name = "PeriodicProducerExpirationCheck", fun = () => {
     lock synchronized {
       producerStateManager.removeExpiredProducers(time.milliseconds)
     }
   }, period = producerIdExpirationCheckIntervalMs, delay = producerIdExpirationCheckIntervalMs, unit = TimeUnit.MILLISECONDS)
 
+  if (producerExpireCheckOption.isEmpty)
+    throw new IllegalStateException("Failed to schedule PeriodicProducerExpirationCheck witch KafkaScheduler.")
```

Comment:
If we get here, it's because we are shutting down the broker. So, it doesn't seem it's an illegal state. Perhaps we could just let it go?